tests/executor/exchange_hash_join_test.cpp

```diff
@@ -0,0 +1,879 @@
+        (join_algorithm != PLAN_NODE_TYPE_EXCHANGE_HASH_JOIN)) {
+      ExpectMoreThanOneTileResults(&left_table_scan_executor,
+                                   left_table_logical_tile_ptrs);
+    } else {
+      ExpectNormalTileResults(BuildTestTableUtil::left_table_tile_group_count,
+                              &left_table_scan_executor,
+                              left_table_logical_tile_ptrs);
+    }
+  }
+
+  // Right scan executor returns logical tiles from the right table
+
+  EXPECT_CALL(right_table_scan_executor, DInit()).WillOnce(Return(true));
+
+  //===--------------------------------------------------------------------===//
+  // Setup right table
+  //===--------------------------------------------------------------------===//
+
+  if (join_test_type == BASIC_TEST || join_test_type == COMPLICATED_TEST ||
+      join_test_type == SPEED_TEST ||
+      join_test_type == LargeTableCorrectnessTest) {
+    ExpectNormalTileResults(BuildTestTableUtil::right_table_tile_group_count,
+                            &right_table_scan_executor,
+                            right_table_logical_tile_ptrs);
+
+  } else if (join_test_type == BOTH_TABLES_EMPTY) {
+    ExpectEmptyTileResult(&right_table_scan_executor);
+
+  } else if (join_test_type == LEFT_TABLE_EMPTY) {
+    if (join_type == JOIN_TYPE_INNER || join_type == JOIN_TYPE_LEFT) {
+      // For hash join, we always build the hash table from right child
+      if ((join_algorithm == PLAN_NODE_TYPE_HASHJOIN ||
+           join_algorithm == PLAN_NODE_TYPE_EXCHANGE_HASH_JOIN)) {
+        ExpectNormalTileResults(
+            BuildTestTableUtil::right_table_tile_group_count,
+            &right_table_scan_executor, right_table_logical_tile_ptrs);
+      } else {
+        ExpectMoreThanOneTileResults(&right_table_scan_executor,
+                                     right_table_logical_tile_ptrs);
+      }
+
+    } else if ((join_type == JOIN_TYPE_OUTER || join_type == JOIN_TYPE_RIGHT)) {
+      ExpectNormalTileResults(BuildTestTableUtil::right_table_tile_group_count,
+                              &right_table_scan_executor,
+                              right_table_logical_tile_ptrs);
+    }
+  } else if (join_test_type == RIGHT_TABLE_EMPTY) {
+    ExpectEmptyTileResult(&right_table_scan_executor);
+  }
+
+  //===--------------------------------------------------------------------===//
+  // Setup join plan nodes and executors and run them
+  //===--------------------------------------------------------------------===//
+
+  oid_t result_tuple_count = 0;
+  oid_t tuples_with_null = 0;
+  auto projection = JoinTestsUtil::CreateProjection();
+  // setup the projection schema
+  auto schema = CreateJoinSchema();
+
+  // Construct predicate
+  std::unique_ptr<const expression::AbstractExpression> predicate(
+      JoinTestsUtil::CreateJoinPredicate());
+
+  // Differ based on join algorithm
+  switch (join_algorithm) {
```

Comment:
The size of the case clauses in this switch statement make it a bit difficult to read. It might be better to move the contents of the case to helper functions